services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/isl/IslCostSpec.groovy

```diff
@@ -0,0 +1,71 @@
+package org.openkilda.functionaltests.spec.isl
+
+import static org.junit.Assume.assumeTrue
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.PathHelper
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.info.event.IslChangeType
+import org.openkilda.messaging.info.event.PathNode
+import org.openkilda.testing.model.topology.TopologyDefinition
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+import org.openkilda.testing.service.aswitch.ASwitchService
+import org.openkilda.testing.service.aswitch.model.ASwitchFlow
+import org.openkilda.testing.service.topology.TopologyEngineService
+import org.openkilda.testing.tools.IslUtils
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+
+class IslCostSpec extends BaseSpecification {
+    @Autowired
+    TopologyDefinition topology
+    @Autowired
+    TopologyEngineService topologyEngineService
+    @Autowired
+    IslUtils islUtils
+    @Autowired
+    PathHelper pathHelper
+    @Autowired
+    ASwitchService aswitch
+
+    @Value('${discovery.interval}')
+    int discoveryInterval
+    @Value('${discovery.timeout}')
+    int discoveryTimeout
+
+    def "ISL cost is NOT increased due to connection loss between switches (not port down)"() {
+        given: "Two active switches with link going through a-switch"
+        def switches = topology.getActiveSwitches()
+        List<List<PathNode>> allPaths = []
+        List<PathNode> aswitchPath
+
+        [switches, switches].combinations().findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            allPaths = topologyEngineService.getPaths(src.dpId, dst.dpId)*.path
+            aswitchPath = allPaths.find { pathHelper.getInvolvedIsls(it).find { it.aswitch } }
+            aswitchPath
+        } ?: assumeTrue("Wasn't able to find suitable switch pair", false)
+
+        when: "Remove a-switch rules to break link between switches"
+        int islCost = pathHelper.getCost(aswitchPath)
```

Comment:
you are naming it 'islCost', but taking the cost of the whole path. should be `int islCost = pathHelper.getCost(aswitchIsl)`

Reply:
by the way, it is probably makes sense to move getCost(Isl isl) method to islUtils intead of pathHelper.